src/main/java/org/apache/commons/lang3/builder/ToStringStyle.java

```diff
@@ -2568,13 +2573,13 @@ protected void appendDetail(StringBuffer buffer, String fieldName, Object value)
                 return;
             }
 
-            if (value.getClass() == String.class) {
+            if (value instanceof Number || value.getClass() == Boolean.class) {
 
-                appendValueAsString(buffer, (String)value);
+                buffer.append(value);
                 return;
             }
 
-            buffer.append(value);
+            appendValueAsString(buffer, value.toString());
```

Comment:
I don't think this is correct for nested object structures. Suppose we have the following class structure:

```java
public class Person {
   private String name;
   private Address address;
}

public class Address {
   private String city;
   private String zip;
}
```

Translating this to JavaScript/JSON should look like this if both, Person and Address use the JsonToStringStyle for their equals method:

```JavaScript
{
   "name": "Peter",
   "address": {
      "city": "New York",
      "zip": "10026"
   }
}
```

If we apply this fix, the result will look like this:

```JavaScript
{
   "name": "Peter",
   "address": "{
      "city": "New York",
      "zip": "10026"
   }"
}
```

Note the additional double quotes around the address object.